fhevm/tee_arithmetic_gas.go

```diff
@@ -0,0 +1,53 @@
+package fhevm
+
+import "encoding/hex"
+
+func teeArithmeticGas(op string, environment EVMEnvironment, input []byte) uint64 {
```

Comment:
The `switch` case is a bit ugly in this function, let's change the function argument to:

```suggestion
func teeArithmeticGas(op string, environment EVMEnvironment, input []byte, gasCosts map[tfhe.FheUintType]uint64) uint64 {
```